inferno-ml-server-types/src/Inferno/ML/Server/Types.hs

```diff
@@ -1043,6 +1055,212 @@ data EvaluationEnv gid p = EvaluationEnv
 instance (Arbitrary p) => Arbitrary (EvaluationEnv gid p) where
   arbitrary = genericArbitrary
 
+data RemoteError
+  = CacheSizeExceeded
+  | -- | Either parent model row corresponding to the model version, or the
+    -- the requested model version itself, does not exist
+    NoSuchModel (Either (Id (Model (EntityId GId))) (Id (ModelVersion (EntityId GId) Oid)))
+  | NoSuchScript VCObjectHash
+  | NoSuchParameter (Id (InferenceParam (EntityId GId) PID))
+  | InvalidScript Text
+  | InvalidOutput Text
+  | -- | Any error condition returned by Inferno script evaluation
+    InfernoError SomeInfernoError
+  | NoBridgeSaved
+  | ScriptTimeout Int
+  | ClientError String
+  | OtherRemoteError Text
+  deriving stock (Show, Eq, Generic)
+  deriving anyclass (ToJSON, FromJSON)
+
+newtype SomeInfernoError = SomeInfernoError String
+  deriving stock (Show, Eq, Generic)
+  deriving anyclass (Exception, ToJSON, FromJSON)
```

Comment:
Changed from existential over `Show` because existential is not trivially deserializable and not really needed here